test/test_test_server.py

```diff
@@ -0,0 +1,11 @@
+import pytest
+
+
+@pytest.mark.asyncio
+async def test_test_server_response(test_server, test_requests):
+    for test_request in test_requests:
+        # Get the response from the test server
+        test_response = await test_server.get_response(test_request.request)
+        # Verify that the response is as expected
+        for attr, val in test_request.attrs.items():
+            assert getattr(test_response, attr) == val
```

Comment:
1. Why there's only one test?
2. The test is not testing the "mocking" behavior. Try to actually write a end to end test that checks that our mocked class is working as expected. for that, you can create an API on our FastAPI application, then mock it inside engine, and then write a test that send request to there, and then request to fastapi, and assert that everything inside the Response object is same.